hummingbot/connector/exchange/okex/okex_api_order_book_data_source.py

```diff
@@ -0,0 +1,312 @@
+
+        while True:
+            try:
+                trading_pairs: List[str] = self._trading_pairs
+                async with websockets.connect(OKCOIN_WS_URI) as ws:
+                    ws: websockets.WebSocketClientProtocol = ws
+
+                    for trading_pair in trading_pairs:
+                        subscribe_request: Dict[str, Any] = {
+                            "op": "subscribe",
+                            "args": [f"spot/trade:{trading_pair}"]
+                        }
+                        await ws.send(json.dumps(subscribe_request))
+
+                    async for raw_msg in self._inner_messages(ws):
+                        # OKEx compresses their ws data
+                        decoded_msg: str = inflate(raw_msg)
+
+                        self.logger().debug("decode menssage:" + decoded_msg)
+
+                        if '"event":"subscribe"' in decoded_msg:
+                            self.logger().debug(f"Subscribed to channel, full message: {decoded_msg}")
+                        elif '"table":"spot/trade"' in decoded_msg:
+                            self.logger().debug(f"Received new trade: {decoded_msg}")
+
+                            for data in json.loads(decoded_msg)['data']:
+                                trading_pair = data['instrument_id']
+                                trade_message: OrderBookMessage = OkexOrderBook.trade_message_from_exchange(
+                                    data, __class__.iso_to_timestamp(data['timestamp']), metadata={"trading_pair": trading_pair}
+                                )
+                                self.logger().debug(f"Putting msg in queue: {str(trade_message)}")
+                                output.put_nowait(trade_message)
+                        else:
+                            self.logger().debug(f"Unrecognized message received from OKEx websocket: {decoded_msg}")
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().error("Unexpected error with WebSocket connection. Retrying after 30 seconds...",
+                                    exc_info=True)
+                await asyncio.sleep(30.0)
+
+    async def _inner_messages(self,
+                              ws: websockets.WebSocketClientProtocol) -> AsyncIterable[str]:
+        # Terminate the recv() loop as soon as the next message timed out, so the outer loop can reconnect.
+        try:
+            while True:
+                try:
+                    msg: str = await asyncio.wait_for(ws.recv(), timeout=self.MESSAGE_TIMEOUT)
+                    yield msg
+                except asyncio.TimeoutError:
+                    # try:
+                    pong_waiter = await ws.ping()
+                    await asyncio.wait_for(pong_waiter, timeout=self.PING_TIMEOUT)
+                    # removed this try as it would fail anyway, #TODO check this
+                    # except asyncio.TimeoutError:
```

Comment:
Yep, you can remove this try except.

Reply:
I think that is referring to the try except that was commented out.
Both actual try except on line 226 and 228 should remain according to instruction from the documentation page.

Reply:
Yeah, that's what i meant, to remove the commented try-except.